app/controllers/admin/events_controller.rb

```diff
@@ -2,7 +2,7 @@
 
 module Admin
   class EventsController < ApplicationController
-    before_action :set_event, only: %i[show edit update destroy]
+    before_action :set_event, :update_event_status, only: %i[show edit update destroy]
```

Comment:
Few things:

1. We don't want to do all this work any time these routes are hit. Instead, use ActiveJob to schedule a disabling of the event on its datetime.
2. Even if we were to keep it this way, we'd need to perform this on ALL event actions including index for all types of users.

I'd just make a collection instance variable @active_event_logs in the index action that queries for all event logs that aren't disabled.

Reply:
Due to the complexity of implementing job scheduling, we chose an alternative way to implement this without having to do N+1 work for every route.